Sources/SwiftParser/Declarations.swift

```diff
@@ -1102,6 +1120,13 @@ extension Parser {
         arena: self.arena)
     }
 
+    // Detect an attempt to use a type parameter pack.
+    var unexpectedAfterName: RawUnexpectedNodesSyntax?
+    if let ellipsis = tryConsumeEllipsisPrefix() {
+      unexpectedAfterName = RawUnexpectedNodesSyntax(
+        elements: [RawSyntax(ellipsis)], arena: self.arena)
+    }
+
```

Comment:
There is a convenience initializer on `RawUnexpectedNodesSyntax` that fails if it is passed only nil elements, so this can be simplified to 

```swift
let ellipsis = tryConsumeEllipsisPrefix()
```

and then below where you are currently passing `unexpectedAfterName` to the `RawAssociatedtypeDeclSyntax` initializer: 

```swift
RawUnexpectedNodesSyntax(ellipsis, arena: self.arena)
```

Reply:
Neat!